flashpay/apps/account/authentication.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any, Dict, Optional, Tuple
+
+from rest_framework_simplejwt.authentication import JWTAuthentication
+from rest_framework_simplejwt.exceptions import InvalidToken
+from rest_framework_simplejwt.settings import api_settings
+
+from rest_framework.authentication import BaseAuthentication
+from rest_framework.exceptions import AuthenticationFailed
+from rest_framework.request import Request
+
+from flashpay.apps.account.models import Account, APIKey
+
+
+class PublicKeyAuthentication(BaseAuthentication):
+    def authenticate(self, request: Request) -> Optional[Tuple[Any, Any]]:
+        public_key = request.META.get("HTTP-X-PUBLIC-KEY")
+        if not public_key:
+            return None
+        try:
+            api_key = APIKey.objects.select_related("account").get(public_key=public_key)
+            account = api_key.account
+            request.network = api_key.network  # type: ignore
+        except APIKey.DoesNotExist:
+            raise AuthenticationFailed("Invalid public key provided")
+        return (account, None)
+
+
+class SecretKeyAuthentication(BaseAuthentication):
+    def authenticate(self, request: Request) -> Optional[Tuple[Any, Any]]:
+        secret_key = request.META.get("HTTP_X_SECRET_KEY")
```

Comment:
Was this meant to be `HTTP-X-SECRET-KEY`?

Reply:
That's correct. it was tested